### PR TITLE
Adjust decoy strategy and fdr_level flag

### PR DESCRIPTION
**Pull request by jonasscheid**
This addresses #228 and #255
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mhcquant/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/mhcquant _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -129,7 +129,7 @@ process {
                 "-enzyme no_enzyme",
                 "-subset_max_train ${params.subset_max_train}",
                 "-doc ${params.description_correct_features} ",
-                (params.fdr_level != 'psm-level-fdrs') ? "-" + params.fdr_level : ""
+                (params.fdr_level != 'psm_level_fdrs') ? "-" + params.fdr_level : ""
```
Comment (marissaDubbelaar) [inline conf/modules.config:132]: check that if this parameter is given that the right annotation is passed to `OPENMS_PERCOLATORADAPTER`, which is still `psm-level-fdrs` as far as I know

File: conf/modules.config [hunk 2]
```diff
@@ -228,7 +228,7 @@ process {
                     "-enzyme no_enzyme",
                     "-subset_max_train ${params.subset_max_train}",
                     "-doc ${params.description_correct_features} ",
-                    (params.fdr_level != 'psm-level-fdrs') ? "-" + params.fdr_level : ""
+                    (params.fdr_level != 'psm_level_fdrs') ? "-" + params.fdr_level : ""
```
Comment (marissaDubbelaar) [inline conf/modules.config:231]: Same here

Comment (jonasscheid) [reply conf/modules.config:231]: see above

File: conf/modules.config (hunk 1 above)
Comment (marissaDubbelaar) [reply conf/modules.config:132]: Then it is fine 👍

**Comment by jonasscheid** [comment]
With the new decoy database, one test dataset seems to contain no features. Should we change the input for the default test also to the [pride](https://github.com/nf-core/test-datasets/blob/mhcquant/sample_sheet_full.tsv) files? @marissaDubbelaar

**Comment by marissaDubbelaar** [comment]
That might be better, let's do it properly then 😉

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,14 +3,17 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## v2.4.1 nfcore/mhcquant "Maroon Gold Boxer" (patch) - yyyy/mm/dd
+## v2.4.1 nfcore/mhcquant "Maroon Gold Boxer" (patch) - 2023/03/24
```
Comment (marissaDubbelaar) [inline CHANGELOG.md:6]: Date might need to change

File: conf/test.config [hunk 4]
```diff
@@ -20,7 +20,7 @@ params {
     max_time   = '6.h'
 
     // Input data
-    fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/mhcquant/test.fasta'
-    input = 'https://raw.githubusercontent.com/nf-core/test-datasets/mhcquant/sample_sheet.tsv'
-    allele_sheet = 'https://raw.githubusercontent.com/nf-core/test-datasets/mhcquant/allele_sheet.tsv'
+    fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/mhcquant/testdata/UP000005640_9606.fasta'
```
Comment (marissaDubbelaar) [inline conf/test.config:23]: Is this the whole human proteome or just a subset?

**Review by marissaDubbelaar — APPROVED** [review]

File: conf/test.config (hunk 4 above)
Comment (jonasscheid) [reply conf/test.config:23]: Whole

**Pull request merged**